src/utils/options.ts

```diff
@@ -0,0 +1,188 @@
+import { PrintConfig } from './printing'
+import { promises } from 'fs'
+import { isAbsolute, join } from 'path'
+import { ok } from './assert'
+import chalk from 'chalk'
+import makeDebug from 'debug'
+
+const debug = makeDebug('kiosk-browser:options')
+
+export interface Options {
+  url: URL
+  autoconfigurePrintConfig?: PrintConfig
+}
+
+export interface Help {
+  help: true
+}
+
+export interface Invalid {
+  error: Error
+}
+
+/**
+ * Gets the URL to navigate.
+ */
+export default async function parseOptions(
+  argv: typeof process.argv = [],
+  env: typeof process.env = {},
+): Promise<Options | Help | Invalid> {
+  debug('parsing options from argv=%o and env=%O', argv, env)
+
+  let urlArg: string | undefined
+  let autoconfiurePrintConfigArg: string | undefined
+  let helpArg: string | undefined
+
+  for (let i = 0; i < argv.length; i++) {
+    const arg = argv[i]
+
+    if (arg === '--url' || arg === '-u') {
```

Comment:
there isn't a nice `argv` parser out there for us to reuse?

Reply:
Oh there are tons, I just don't like them. I can try [`yargs`](https://www.npmjs.com/package/yargs) which seems to be one of the most popular if you like.